siuba/experimental/pivot/test_pivot.py

```diff
@@ -7,14 +7,19 @@
 # TODO: pivot_longer rows in different order from gather. Currently using
 #       a row order invariant form of test. Should fix to conform to pivot_longer.
 # TODO: need to be careful about index. Should preserve original indices for rows.
+# TODO: Current behaviour of pivot_longer is to preserve the index. Therefore
+#       it needs to be reset for any assertion.
```

Comment:
I guess this is more of a note than a TODO, but goes in line with your TODO one up. In your screencast you mentioned that it might be a good idea to retain the initial indexes the Dataframe has prior to pivoting and I think I agree that this makes sense overall. However, anywhere that uses `assert_equal_frame` will require identical indexes, so they need to be reset for that comparison to work.

Reply:
Ah, good point--is the main issue that "target" data frames like...

```
dst = data_frame(name = ["x", "y", "x", "y"], value = [1, 3, 2, 4])
```

don't have the right index defined? I wouldn't mind allowing `data_frame` to set set an index using  either an `_index` or `index` argument, if it makes it easier!

Reply:
Yes, you hit the nail on the head. Rewriting `data_frame` to allow for an optional `_index` would allow to set that here in one call, without having to resort to `reset_index` on the side of `pv` or (equally as bad) `reindex` on the side of `dst`.

Reply:
I added merged the change to add a special `_index` argument to `data_frame` into master--in case it makes things easier! (definitely will need it down the road :)